pkg/asset/installconfig/azure/client.go

```diff
@@ -274,9 +275,21 @@ func (c *Client) GetDiskEncryptionSet(ctx context.Context, subscriptionID, group
 	return &diskEncryptionSet, nil
 }
 
+func (c *Client) GetVMCapabilities(ctx context.Context, instanceType, region string) (map[string]string, error) {
+	typeMeta, err := c.GetVirtualMachineSku(ctx, instanceType, region)
+	if err != nil {
+		return nil, fmt.Errorf("error onnecting to Azure client: %s", err.Error())
+	}
```

Comment:
You need to check that `typeMeta` is a valid pointer since the passed `instanceType` might not be available in the specified region.